kinesis_firehose/main.tf

```diff
@@ -0,0 +1,65 @@
+# Create tags
+module "tags" {
+  source = "github.com/variant-inc/lazy-terraform//submodules/tags?ref=v1"
+  # source = "../submodules/tags" # For testing
+  user_tags    = var.user_tags
+  octopus_tags = var.octopus_tags
+  name         = var.kinesis_firehose_delivery_stream
+}
+
+# Create resource for kinesis firehose delivery stream
+resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
+  name        = var.kinesis_firehose_delivery_stream
+  tags        = module.tags.tags
+  destination = "extended_s3"
+  server_side_encryption {
+    enabled = true
+  }
+
+  extended_s3_configuration {
+    role_arn            = aws_iam_role.firehose_role.arn
+    bucket_arn          = aws_s3_bucket.bucket.arn
+    prefix              = var.firehose_s3.bucket_content_prefix
+    error_output_prefix = var.firehose_s3.bucket_content_error_prefix
+    buffer_size         = var.firehose_s3.buffer_size
+    buffer_interval     = var.firehose_s3.buffer_interval
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = "log_group-${var.kinesis_firehose_delivery_stream}"
+      log_stream_name = "log_stream-${var.kinesis_firehose_delivery_stream}"
+    }
+
+    processing_configuration {
+      enabled = "true"
+
+      processors {
+        type = "Lambda"
+
+        parameters {
+          parameter_name  = "LambdaArn"
+          parameter_value = "${aws_lambda_function.lambda_processor.arn}:$LATEST"
+        }
+      }
+    }
+  }
+}
+
+# Create resource S3 bucket where firehose will stream records
+resource "aws_s3_bucket" "bucket" {
```

Comment:
Use the s3 module created by us if you want replication

Reply:
In this case replication is not needed

Reply:
still there

Reply:
Just rememebered, our S3 module disables public and S3 access. So, I'd suggest using that.